File: weather_script.js
const fetch = require('node-fetch');

const API_KEY = '81666199fe30e191bd70f701b2b0ff34'; // Reemplaza con tu clave de API
const BASE_URL = 'https://api.openweathermap.org/data/2.5/weather';

async function getWeather(city) {
    const url = `${BASE_URL}?q=${city}&appid=${API_KEY}&units=metric`;
    try {
        const response = await fetch(url);
        if (!response.ok) {
            throw new Error('Error al recuperar los datos del clima');
        }
        const data = await response.json();
        return data;
    } catch (error) {
        console.error('Error:', error);
    }
}

async function displayWeather(city) {
    const weatherData = await getWeather(city);
    if (weatherData) {
        console.log(`El clima en ${city}:`);
        console.log(`Temperatura: ${weatherData.main.temp}°C`);
        console.log(`Humedad: ${weatherData.main.humidity}%`);
        console.log(`Descripción: ${weatherData.weather[0].description}`);
    }
}

// Ejemplo de uso
displayWeather('Madrid');

File: weather_script.py
import requests

def get_weather(api_key, city):
    base_url = "http://api.openweathermap.org/data/2.5/weather"
    params = {
        'q': city,
        'appid': api_key,
        'units': 'metric'
    }
    response = requests.get(base_url, params=params)
    if response.status_code == 200:
        return response.json()
    else:
        return None

def main():
    api_key = 'YOUR_API_KEY'  # Replace with your OpenWeatherMap API key
    city = 'London'  # Replace with the city you want to get the weather for
    weather_data = get_weather(api_key, city)
    if weather_data:
        print(f"City: {weather_data['name']}")
        print(f"Temperature: {weather_data['main']['temp']}°C")
        print(f"Weather: {weather_data['weather'][0]['description']}")
    else:
        print("Failed to retrieve weather data")

if __name__ == "__main__":
    main()